inst/include/dust/utils.hpp

```diff
@@ -64,6 +64,37 @@ HOSTDEVICE T epsilon() {
 #endif
 }
 
+__nv_exec_check_disable__
+inline int integer_max() {
+#ifdef __CUDA_ARCH__
+  return INT_MAX;
+#else
+  return std::numeric_limits<int>::max();
+#endif
+}
+
+template <typename T>
+T lgamma(T x);
```

Comment:
Not sure how host/device works in this case – is this prototype needed?
I think that this: https://github.com/mrc-ide/dust/blob/master/inst/include/dust/xoshiro.hpp#L142-L169
is the closest we have to this logic in the existing code